compiler/rustc_middle/src/middle/stability.rs

```diff
@@ -271,7 +283,13 @@ pub enum EvalResult {
     Allow,
     /// We cannot use the item because it is unstable and we did not provide the
     /// corresponding feature gate.
-    Deny { feature: Symbol, reason: Option<Symbol>, issue: Option<NonZeroU32>, is_soft: bool },
+    Deny {
+        feature: Symbol,
+        reason: Option<Symbol>,
+        issue: Option<NonZeroU32>,
+        is_soft: bool,
+        using_vec_allocator_api: bool,
```

Comment:
I'd prefer that we avoid adding feature gate-specific fields here, like `using_vec_allocator_api`, and instead add some generic fields like `suggestion: Option<Span, String, Applicability>`.